### ISSUE TITLE
Feature Request : Webhooks

### ISSUE THREAD

**Issue by gpshex007** [traccar/traccar#4492]
Feature request to support Webhooks:

Ability to add webhook and send real-time events within traccar to external services.

Basic idea is to provide users the ability to define an endpoint URL that should receive information based on specific events/alarms (configurable by user)

Difference between forward event/position and webhook (forward option available at the moment is limited to admin configuring it and also it is limited to sending to one service/url)

Webhook on the other hand can be a feature within traccar that users can take benefit of.

**Comment by tananaev** [comment]
What's the business case for it?

**Comment by gpshex007** [comment]
Plenty of possibilities exist with webhooks.

Someone running traccar as as service can offer their users (businesses) the option to configure end point URLS to which data would be sent based on event types the user selects.

The webhooks will allow these business/users to receive real-time data into their CRM/infrastrcuture or as a matter of fact even allow them to have specific actions to take place based on the data received.

Use case:  A business with 100's of vehicles could develop a dashboard for their team to view, and configure the webhook on traccar and collect these data in real time.

Or a business/user can use the webhooks to trigger or activate certain machinery / switches / or home automation related objects by configuring webhooks to their custom services/applications.

Basically this will help add a feature to traccar  where traccar users can provide their customers additional integration options with external applications/devices.

**Comment by oleghalin** [comment]
Are there any plans to integrate Webhooks?

**Comment by tananaev** [comment]
Traccar already has webhooks, but global and not per user. It should be possible to split it using an external service.
